### PR TITLE
Instead of returning an error we have to set the shutdown_result_, th…

### PR DESCRIPTION
**Pull request by jwillemsen**
…is method is called from the ACE reactor which is not aware of the DDS return codes

    * dds/DCPS/DomainParticipantImpl.cpp:

### CODE REVIEW

**Comment by jwillemsen** [comment]
Part of #506

**Comment by mitza-oci** [comment]
retest this please
(starting Jenkins build)

File: dds/DCPS/DomainParticipantImpl.cpp [hunk 1]
```diff
@@ -2187,16 +2189,16 @@ DomainParticipantImpl::handle_exception(ACE_HANDLE /*fd*/)
       = pubPtr->delete_contained_entities();
 
       if (result != DDS::RETCODE_OK) {
-        return result;
+        ret = result;
```
Comment (mitza-oci) [inline dds/DCPS/DomainParticipantImpl.cpp]: Before this change, a non-OK result stopped further processing.  Now it's proceeding with the next steps.  Is that what we want?

File: dds/DCPS/DomainParticipantImpl.cpp [hunk 2]
```diff
@@ -2219,20 +2221,19 @@ DomainParticipantImpl::handle_exception(ACE_HANDLE /*fd*/)
       DDS::ReturnCode_t result = subPtr->delete_contained_entities();
 
       if (result != DDS::RETCODE_OK) {
-        return result;
+        ret = result;;
```
Comment (mitza-oci) [inline dds/DCPS/DomainParticipantImpl.cpp]: This line has two ;s (also below on 2230)

File: dds/DCPS/DomainParticipantImpl.cpp (hunk 1 above)
Comment (jwillemsen) [inline dds/DCPS/DomainParticipantImpl.cpp]: Not 100% sure, maybe the next steps will result in other errors but this gives the easiest control flow

**Comment by jrw972** [comment]
retest this please

**Comment by jrw972** [comment]
retest this please

**Comment by mitza-oci** [comment]
@huangminghuang please look at this change to DomainParticipantImpl (not necessarily other files) and see if it can be combined with what you're working on

**Pull request merged**